tutorials/add_model.md

```diff
@@ -0,0 +1,139 @@
+# Contributing a model to Cornac
+
+This tutorial describes how to integrate a recommender model into Cornac. We assume that you have already forked the Cornac repository to your own account.
+
+## Directory & file structure
+
+For convenience assume that the model of interest is PMF. At this point it is useful to recall that Cornac follows the [PEP 8](https://www.python.org/dev/peps/pep-0008/) coding conventions. One good way to get started is to update the directory and file structure as follows.
+```
+cornac    
+│
+└───cornac
+│   │
+│   └───models
+│       │   __init__.py
+│       │   recommender.py
+│       │
+│       └───pmf
+│           │   __init__.py
+│           │   recom_pmf.py
+│           │   requirements.txt
+```
+Note that you only need to add the `pmf` branch as the rest of the structure is already in place.
+
+## Creating a Cornac model in 4 steps
+
+### 1. Extending the Recommender class
+
+The main file is `recom_pmf.py`. The name of this file should always start with the prefix `recom_`. Here is the minimal structure of such file:
```

Comment:
The main file is `recom_pmf.py` where some required APIs for a recommender model will be implemented. The name of this file should always start with the prefix `recom_`. Here is the minimal structure of such file:

Reply:
Noted, thanks!